Graviton_monitoring_differences.md

```diff
@@ -0,0 +1,56 @@
+# AWS Graviton monitoring differences
+Processor architecture choices are made for various reasons such as compatibility, performance and newer application domain support. These choices imply that the tools supported on the processors would differ too. Listed below are some  differences in the tools supported on AWS Gravtion when compared to x86 processor architectures.
+
+## Utilities for obtaining processor stats
+|Processor	|x86	|Graviton2,3	|
+|---	|---	|---	|
+|CPU frequency listing	|*lscpu, /proc/cpuinfo, dmidecode*	|*dmidecode*	|
+|*turbostat* support	|Yes	|No	|
+|*i7z* Works	|Yes	|No	|
+|Performance monitoring tools	|_[VTune Profiler](https://www.intel.com/content/www/us/en/developer/tools/oneapi/vtune-profiler.html) and [PCM](https://github.com/opcm/pcm), [Linux perf](https://www.brendangregg.com/perf.html)_	|_[Linux perf](https://www.brendangregg.com/perf.html), [Arm Forge](https://developer.arm.com/Tools%20and%20Software/Arm%20Forge)_	|
+
+More details on how to use Linux perf utility on AWS Graviton processors is available [here](https://github.com/aws/aws-graviton-getting-started/blob/main/optimizing.md#profiling-the-code).
+
+**Note:** On AWS Graviton 2/3 processors, Turbo isn’t supported. So, utilities such as ‘turbostat’ aren’t supported/ relevant for Arm architecture (and not on AWS Graviton processor either). Also, tools such as *[i7z](https://code.google.com/archive/p/i7z/)* for discovering CPU frequency, turbo, sockets and other information are only supported on Intel architecture/ processors.
+
+## Sample output
+#### Using lscpu utility
+*$ lscpu*
+|Attribute| Value|
+|---    |---    |
+|Architecture| aarch64|
+|CPU(s)| 64|
+|On-line CPU(s) list| 0-63|
+|L1d| 4 MiB (64 instances)|
+|...|...|
+|NUMA node(s)|  1|
+|...|...|
+
+#### Using dmidecode to get CPU frequency info
+*$ sudo dmidecode | less*
+|Attribute| Value|
+|---    |---    |
+|Socket Designation| CPU00|
+|Type| Central Processor|
+|Family| ARMv8|
+|Manufacturer| AWS|
+|...|...|
+|Max Speed| 2600 MHz|
+|Current Speed| 2600 MHz|
+|...|...|
+
+#### Collect CPU counter statistics for the specified command
```

Comment:
again, introduce perf and talk about the commands.

Reply:
I think perf is introduced in the perfrunbook page and also on Brendan Gregg perf page. So, I kept it brief here.

Reply:
still, we don't have things like intel's PCM and we need to say you can use perf and it has these optios

Reply:
same for intel memory latency checker, lmbench or other

Reply:
please talk about how to use this for someone coming from other tools and what perf can measure. e.g. on metal you can get interconnect counters too.

Reply:
I prefect not to add lmbench and Intel memory latency chencker as the list is a bit unending and we can't cover all the tools on such a page. By definition, the Intel MLC is an Intel tool, so I don't see a point in adding it here. I tried to build lmbench but with some dependency failures and I feel it isn't worth it.

I have added details on how to use Linux perf and a note about PCM.

Reply:
With some fixes/ updates to the package, I could lmbench to build and run successfully. So, there isn't anything unique about it to compare a run on x86 vs Graviton and to document it here.